cannon/cmd/run.go

```diff
@@ -248,7 +248,8 @@ func (p *ProcessPreimageOracle) Close() error {
 func (p *ProcessPreimageOracle) wait() {
 	err := p.cmd.Wait()
 	var waitErr error
-	if err, ok := err.(*exec.ExitError); !ok || !err.Success() {
```

Comment:
This is some fun go trivia.  If `err` is `nil`, then `ok` is false but we then set `waitErr` to the value of the _shadowed_ `err` variable which is in fact `(*exec.ExitError)(nil)` which is non-nil (but will return `<nil>` from `Error()` which makes things particularly confusing).